src/main/java/org/broadinstitute/hellbender/utils/SimpleInterval.java

```diff
@@ -225,10 +225,15 @@ public boolean overlaps( final Locatable other ) {
       * This is the same as plain overlaps if margin=0.
       *
       * @param other interval to check
-      * @param margin how many bases may be between the two interval for us to still consider them overlapping.
+      * @param margin how many bases may be between the two interval for us to still consider them overlapping; must be non-negative
       * @return true if this interval overlaps other, otherwise false
+      * @throws IllegalArgumentException if margin is negative
       */
      public boolean overlapsWithMargin(final Locatable other, final int margin) {
+         if ( margin < 0 ) {
```

Comment:
Why not use `ParamUtils.isPositiveOrZero`?

Reply:
That was the initial implementation, but there's a very reasonable performance concern for string construction in the error message, so changed to `if`